From 3047101050991dca283360853c8f3403501d50d1 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 10 Sep 2014 00:17:42 +0100
Subject: [PATCH 1/2] autotools: Set a default buildDir

So that the un-changed optimisation in setBuildDirectory works
correctly.

Change-Id: Ie31fb2722c663cd0f998458d536181f47c83a37d
---
 src/plugins/autotoolsprojectmanager/autotoolsbuildconfiguration.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/plugins/autotoolsprojectmanager/autotoolsbuildconfiguration.cpp b/src/plugins/autotoolsprojectmanager/autotoolsbuildconfiguration.cpp
index f17b1d7..29d3095 100644
--- a/src/plugins/autotoolsprojectmanager/autotoolsbuildconfiguration.cpp
+++ b/src/plugins/autotoolsprojectmanager/autotoolsbuildconfiguration.cpp
@@ -64,7 +64,11 @@ using namespace ProjectExplorer::Constants;
 //////////////////////////////////////
 AutotoolsBuildConfiguration::AutotoolsBuildConfiguration(Target *parent)
     : BuildConfiguration(parent, Core::Id(AUTOTOOLS_BC_ID))
-{ }
+{
+  // /<foobar> is used so the un-changed check in setBuildDirectory() works correctly.
+  // The leading / is to avoid the relative the path expansion in BuildConfiguration::buildDirectory.
+  BuildConfiguration::setBuildDirectory(Utils::FileName::fromString(QString::fromLatin1("/<foobar>")));
+}
 
 NamedWidget *AutotoolsBuildConfiguration::createConfigWidget()
 {
-- 
2.4.0

